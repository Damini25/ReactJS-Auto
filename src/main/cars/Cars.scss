.cars-main {
    width: 70%;
    padding: 18px;
    height: 90%;
}
.features {
    border: 1px solid #ededed;
    display: flex;
    margin-bottom: 2%;
    border-radius: 2px;
    .carimg {
        width: 20%;
        display: flex;
        border: 1px solid #ededed;
        /* height: 100%; */
        margin: 12px;
        .picture{
            width: 80%;
            padding: 10%;
        }
    }
    .details {
        width: 80%;
        margin: 12px;
        padding-left: 3%;
        p {
            margin: 0px;
            margin-bottom: 2%;
            a {
                text-decoration: none;
            }
        }
    }
}
.pagination{
    display: flex;
    justify-content: center;
    padding-top: 3%;
    .MuiPaginationItem-textPrimary.Mui-selected{
        background-color: #EA7F28;
    }
}