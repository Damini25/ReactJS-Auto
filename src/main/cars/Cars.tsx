import React from 'react';
import './Cars.scss';
import Pagination from '@material-ui/lab/Pagination';
import { Link } from "react-router-dom";
import {capitalize} from '../../utils';

export interface Props {
  availableCars: any[],
  pageCount: number,
  onPageChange: (event: any, value: number) => void,
  selectedPage: number,
  totalCars: number
}
export interface Cars {
  stockNumber: number,
  manufacturerName: string,
  modelName: string,
  mileage: {
    number: number,
    unit: string
  },
  fuelType: string,
  color: string,
  pictureUrl: string
}
function Cars({ availableCars, pageCount = 1, onPageChange, selectedPage = 1, totalCars }: Props) {
  console.log('incars', availableCars)
  const itemsPerPage = selectedPage <= Math.floor(totalCars / 10) ? 10 : (totalCars % 10);
  
  const renderCars = () => {
    return availableCars && availableCars.map((item: Cars, ind) => {
      return (<div className="features" key={ind}>
        <div className="carimg">
          <img src={item.pictureUrl} className="picture" alt="logo" />
        </div>
        <div className="details">
          <p className="font18 fontbold">{`${item.manufacturerName} ${item.modelName}`}</p>
          <p className="font14">{`Stock # ${item.stockNumber} - ${item.mileage.number + ' ' + item.mileage.unit} - ${item.fuelType} - ${capitalize(item.color)} `}</p>
          <p className="font12">
          <Link to={`/cardetails/${item.stockNumber}`} className="ctacolor">View Details</Link>
          </p>
        </div>
      </div>);
    })
  }
  return (
    <div className="cars-main">
      <div>
        <p className="font18 fontbold" data-testid="available-cars-id">Available cars</p>
        <p>Showing {itemsPerPage} of {totalCars} results</p>
      </div>
      {renderCars()}
      <div className="pagination">
        <Pagination count={pageCount} page={selectedPage} onChange={onPageChange} color="primary" showFirstButton showLastButton />
      </div>
    </div>
  );
}

export default Cars;
