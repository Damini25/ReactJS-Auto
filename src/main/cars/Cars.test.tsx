import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Cars from './Cars';

const props = {
    availableCars:[],
    pageCount: 1,
    onPageChange:jest.fn(),
    selectedPage: 1,
    totalCars:10,
}
afterEach(cleanup)
test('render <Cars>', () => {
    const wrapper = render(<Cars {...props} />)
    expect(wrapper).toMatchSnapshot()
})