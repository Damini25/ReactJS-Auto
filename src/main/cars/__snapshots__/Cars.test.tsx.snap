// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Cars> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cars-main"
      >
        <div>
          <p
            class="font18 fontbold"
            data-testid="available-cars-id"
          >
            Available cars
          </p>
          <p>
            Showing 
            10
             of 
            10
             results
          </p>
        </div>
        <div
          class="pagination"
        >
          <nav
            aria-label="pagination navigation"
            class="MuiPagination-root"
          >
            <ul
              class="MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to first page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-current="true"
                  aria-label="page 1"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-selected"
                  tabindex="0"
                  type="button"
                >
                  1
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to last page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cars-main"
    >
      <div>
        <p
          class="font18 fontbold"
          data-testid="available-cars-id"
        >
          Available cars
        </p>
        <p>
          Showing 
          10
           of 
          10
           results
        </p>
      </div>
      <div
        class="pagination"
      >
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root"
        >
          <ul
            class="MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to first page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-current="true"
                aria-label="page 1"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-selected"
                tabindex="0"
                type="button"
              >
                1
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-label="Go to last page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiPaginationItem-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
